### ISSUE TITLE
Issue with rest spread and computed properties using string template

### ISSUE THREAD

**Issue by Nickman87** [babel/babel#9563]
## Bug Report

**Current Behavior**
When using string templates in computed properties, the property name isn't correctly excluded from the rest properties object.
Looking at the compiled code, it puts 'undefined' instead of the actual property name. If you use string concatenation, this does work perfectly.

**Input Code**
[REPL sample](https://babeljs.io/repl/build/master#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBAlmADgV1gXhgbwL4G4BQA9ITAKIAeUATgIbCw0A2jMA1mCAO5gw1TVwARqgCmEGADMqIALYwoACxExkEEVXyhIsTPhgwA5nABuIsAH0wNGSIBchk2YBy1kQBo9MAOSMa0S65e9r7QLjYe-gDaAAagyGDUAJ7RALr2cQlUiREwkYhUIhJw5DAA1N7QfCJeaTCVUO6eMQAkmPmFxdhaagkQqengPVAQOcQwAILM8kowBdCeAHRLIIrq4_xUQqIQ-NgwGAgoULhAA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.3.3)

```js
const input = {};
// Extract all known attributes from the user
const {
  given_name: givenName,
  'last_name': lastName,
  [`country`]: country,
  [prefix + 'state']: state,
  [`${prefix}consents`]: consents,
  // All the rest
  ...otherAttributes
} = input;
```

**Expected behavior/code**
All given computed properties should be excluded from the `otherAttributes` object, but the ones using template strings remain.

**Environment**
- Babel version(s): v7.3.3 (probably older versions too)
- Node/npm version: any
- OS: any

**Possible Solution**
Make sure the transpiler knows what a template string is and correctly copy it over to the exclusion list. Sound like something that is easy to fix, but I have no idea where to start :).

**Comment by nicolo-ribaudo** [comment]
If you (or anyone else) wants to open a PR, this is the file which transforms object rest and spread properties: https://github.com/babel/babel/blob/master/packages/babel-plugin-proposal-object-rest-spread/src/index.js

The problem is that the `extractNormalizedKeys` function (which converts literals to string; e.g. `"foo"`->`"foo"`, `1`->`"1"`, `true`->`"true"`):

https://github.com/babel/babel/blob/9f3457797ff9b90f4cc794c173d626163f9a6046/packages/babel-plugin-proposal-object-rest-spread/src/index.js#L59-L77

It handles templates like it handles simple literals; but templates have a different AST structure. You can see how templates are represented using https://astexplorer.net/.
We need to handle templates differently, and keep in mind that we can optimize only templates without substituitions: any other template could have side effects.

You can add a test for this bug in https://github.com/babel/babel/tree/master/packages/babel-plugin-proposal-object-rest-spread/test/fixtures/object-rest

If you don't know how to clone Babel, follow these steps: (you need to have `make` and `yarn` available on your machine).
1. Write a comment there to know other possible contributors that you are working on this bug.
1. Fork the repo
2. Run `git clone https://github.com/<YOUR_USERNAME>/babel.git && cd babel`
3. Run `yarn && make bootstrap`
4. Wait :hourglass_flowing_sand: 
5. Run `make watch` (or `make build` whenever you change a file)
6. Add a test (only `input.js`; `output.js` will be automatically generated)
7. Update the code!
8. `yarn jest object-rest-spread` to run the tests
    - If some test outputs don't match but the new results are correct, you can delete the bad `output.js` files and run the tests again
9. If it is working, run `make test` tu run all the tests
10. Run `git push` and open a PR!

**Comment by pnowak** [comment]
Hi. I would like to take this issue!

**Comment by nicolo-ribaudo** [comment]
If you have any other question, feel free to contact me or any other team member on Slack!

**Comment by nicolo-ribaudo** [comment]
Fixed by #9628
